stellar/controllers/ethereum_refund_handler.go

```diff
@@ -0,0 +1,70 @@
+package controllers
+
+import (
+	"database/sql"
+	"encoding/json"
+	"net/http"
+
+	"github.com/stellar/go/support/render/problem"
+	"github.com/stellar/starbridge/backend"
+	"github.com/stellar/starbridge/ethereum"
+	"github.com/stellar/starbridge/store"
+)
+
+type EthereumSignatureResponse struct {
+	Address    string `json:"address"`
+	Signature  string `json:"signature"`
+	DepositID  string `json:"depositId"`
+	Expiration int64  `json:"expiration,string"`
+}
+
+type EthereumRefundHandler struct {
+	Observer                ethereum.Observer
+	Store                   *store.DB
+	EthereumRefundValidator backend.EthereumRefundValidator
+	EthereumFinalityBuffer  uint64
+}
+
+func (c *EthereumRefundHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	deposit, err := getEthereumDeposit(c.Observer, c.Store, c.EthereumFinalityBuffer, r)
+	if err != nil {
+		problem.Render(r.Context(), w, err)
+		return
+	}
+
+	// Check if outgoing transaction exists
+	row, err := c.Store.GetEthereumSignature(r.Context(), store.Refund, deposit.ID)
+	if err != nil && err != sql.ErrNoRows {
+		problem.Render(r.Context(), w, err)
+		return
+	}
+	if err == nil {
+		err = json.NewEncoder(w).Encode(EthereumSignatureResponse{
```

Comment:
Nit. It's super unlikely that this will return `err` but if it does it's possible that some bytes will have been already written and it won't be possible to modify the header (line 49). Thus I think it's better to use `json.Marshal` here.